### PR TITLE
Add excusing for absence

### PR DESCRIPTION
**Pull request by kapi2289**
# Excusing for absence

### More info
--> #317 

### How does it work?

GHi6rKK6aT

### CODE REVIEW

File: app/src/main/java/io/github/wulkanowy/ui/modules/attendance/AttendanceItem.kt [hunk 1]
```diff
@@ -26,6 +29,28 @@ class AttendanceItem(val attendance: Attendance) : AbstractFlexibleItem<Attendan
             attendanceItemSubject.text = attendance.subject
             attendanceItemDescription.text = attendance.name
             attendanceItemAlert.visibility = attendance.run { if (absence && !excused) VISIBLE else INVISIBLE }
+            attendanceItemNumber.visibility = GONE
+            attendanceItemExcuseInfo.visibility = GONE
+            attendanceItemExcuseCheckbox.visibility = GONE
```
Comment (mklkj) [inline app/src/main/java/io/github/wulkanowy/ui/modules/attendance/AttendanceItem.kt]: Maybe show checkboxes after clicking some button, e.g FAB?

File: app/src/main/java/io/github/wulkanowy/ui/modules/attendance/AttendancePresenter.kt [hunk 2]
```diff
@@ -82,6 +85,27 @@ class AttendancePresenter @Inject constructor(
         }
     }
 
+    fun onExcuseCheckboxSelect(attendanceItem: Attendance, checked: Boolean) {
+        attendanceToExcuseList = when (checked) {
```
Comment (mklkj) [inline app/src/main/java/io/github/wulkanowy/ui/modules/attendance/AttendancePresenter.kt]: You can simply use if stmt

File: app/build.gradle [hunk 3]
```diff
@@ -115,14 +115,15 @@ ext {
     dagger = "2.25.3"
     chucker = "2.0.4"
     mockk = "1.9.2"
+    mockito_core = "3.0.11"
```
Comment (mklkj) [inline app/build.gradle]: Remove this line

Comment (mklkj) [inline app/build.gradle]: Remove this line

File: app/build.gradle [hunk 4]
```diff
@@ -115,14 +115,15 @@ ext {
...
 }
 
 configurations.all {
     resolutionStrategy.force "androidx.constraintlayout:constraintlayout:1.1.3"
 }
 
 dependencies {
-    implementation "io.github.wulkanowy:sdk:0.14.0"
+    implementation "io.github.wulkanowy:sdk:d035e66"
```
Comment (mklkj) [inline app/build.gradle]: Please update to `c4d8c21`

File: app/src/main/res/menu/context_menu_excuse.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+
+    <item
+        android:id="@+id/excuseMenuSubmit"
+        android:icon="@drawable/ic_all_done"
+        android:title="@string/attendance_excuse_dialog_submit"
+
```
Comment (Faierbel) [inline app/src/main/res/menu/context_menu_excuse.xml]: usuń tą przerwę

**Pull request merged**